Tests/test_core_resources.py

```diff
@@ -177,13 +177,14 @@ def test_units(self):
         Image._apply_env_variables({"PILLOW_BLOCK_SIZE": "2m"})
         assert Image.core.get_block_size() == 2 * 1024 * 1024
 
-    def test_warnings(self):
-        pytest.warns(
-            UserWarning, Image._apply_env_variables, {"PILLOW_ALIGNMENT": "15"}
-        )
-        pytest.warns(
-            UserWarning, Image._apply_env_variables, {"PILLOW_BLOCK_SIZE": "1024"}
-        )
-        pytest.warns(
-            UserWarning, Image._apply_env_variables, {"PILLOW_BLOCKS_MAX": "wat"}
-        )
+    @pytest.mark.parametrize(
+        "vars",
+        [
+            {"PILLOW_ALIGNMENT": "15"},
+            {"PILLOW_BLOCK_SIZE": "1024"},
+            {"PILLOW_BLOCKS_MAX": "wat"},
+        ],
+    )
+    def test_warnings(self, vars):
+        with pytest.warns(UserWarning):
+            Image._apply_env_variables(vars)
```

Comment:
Singular, only applying one at a time:
```suggestion
        "var",
        [
            {"PILLOW_ALIGNMENT": "15"},
            {"PILLOW_BLOCK_SIZE": "1024"},
            {"PILLOW_BLOCKS_MAX": "wat"},
        ],
    )
    def test_warnings(self, var):
        with pytest.warns(UserWarning):
            Image._apply_env_variables(var)
```